include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -602,6 +752,246 @@ struct __merge_fn
 
 inline constexpr __internal::__merge_fn merge;
 
+namespace __internal
+{
+
+struct __fill_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _T = std::ranges::range_value_t<_R>>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> &&
+             std::ranges::output_range<_R, const _T&> && std::ranges::sized_range<_R>
```

Comment:
`std::ranges::output_range<_R, const _T&>` does not appear in the spec requirements for `fill`.  Should it?

Reply:
@akukanov, @rarutyun 
We really have one more discrepancy between oneDPL spec ()  and p3179 proposal (https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3179r4.html#modify_fill) - `std::indirectly_writable` (in oneDPL spec) instead of `std::ranges::output_range`. 
It was missed or it was done intentionally?

Reply:
Discussed with Ruslan offline. Yes, it was done intentionally.  oneDPL spec for parallel ranges based `fill `and `generate` has less strict requirement - `std::indirectly_writable` instead of `std::ranges::output_range`.
I will change the code according to the spec.
Thank you, Dan.

Reply:
Could you describe why `fill` (accepts a range) has `output_range<R, const T&>` and `fill_n` (accepts an iterator) has `indirectly_writable<O, const T&>` in [the proposal](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3179r4.html#modify_fill)?
Could you also confirm again that we should use `std::indirectly_writable` with the range-based `fill`?

Reply:
I think the [most recent revision](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2025/p3179r7.html#modify_fill) uses `std::indirectly_writable` for `fill` as well.  

I cant really speak to why `fill_n` uses an iterator rather than a range as output.

Reply:
> I think the [most recent revision](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2025/p3179r7.html#modify_fill) uses std::indirectly_writable for fill as well.

Thanks, that clarifies it for me.